hazelcast-jet-core/src/main/java/com/hazelcast/jet/Sinks.java

```diff
@@ -58,6 +65,165 @@ private Sinks() {
         return new SinkImpl<>(sinkName, metaSupplier);
     }
 
+    /**
+     * Returns a sink that uses the supplied functions to extract the key
+     * and value with which to update a Hazelcast {@code IMap}. If the map
+     * already contains the key, it applies the given {@code mergeFn} to
+     * resolve the existing and the proposed value into the value to use. If
+     * the value comes out as {@code null}, it removes the key from the map.
+     * Expressed as code, the sink performs the equivalent of the following for
+     * each item:
+     * <pre>
+     * K key = toKeyFn.apply(item);
+     * V oldValue = map.get(key);
+     * V newValue = toValueFn.apply(item);
+     * V resolved = (oldValue == null)
+     *            ? newValue
+                  : mergeFn.apply(oldValue, newValue);
+     * if (value == null)
+     *     map.remove(key);
+     * else
+     *     map.put(key, value);
+     * </pre>
+     *
+     * @param mapName   name of the map
+     * @param toKeyFn   function that extracts the key from the input item
+     * @param toValueFn function that extracts the value from the input item
+     * @param mergeFn   function that merges the existing value with the value acquired from the
+     *                  received item
+     * @param <E> input item type
+     * @param <K> key type
+     * @param <V> value type
+     */
+    public static <E, K, V> Sink<E> mapWithMerging(
+            @Nonnull String mapName,
+            @Nonnull DistributedFunction<E, K> toKeyFn,
+            @Nonnull DistributedFunction<E, V> toValueFn,
+            @Nonnull DistributedBinaryOperator<V> mergeFn
+    ) {
+        return fromProcessor("mapWithMerging(" + mapName + ')',
+                mergeMapP(mapName, toKeyFn, toValueFn,  mergeFn));
+    }
+
+    /**
+     * Returns a sink equivalent to {@link #mapWithMerging}, but for a map
+     * in a remote Hazelcast cluster identified by the supplied {@code
+     * ClientConfig}.
+     */
```

Comment:
Document the possibility to configure `INVOCATION_TIMEOUT_SECONDS`.

Reply:
I think this is not needed here as this goes for all remote sinks. We could refer to it in the ref manual in the Hazelcast connectors section.

Reply:
I meant documenting it at all methods. I personally never go to refman for such details. In refman I'm mostly looking for concepts and introductions, the rest should be in javadoc, which I read as necessary.